libraries/CurieBLE/src/internal/BLEDeviceManager.cpp

```diff
@@ -1370,7 +1370,7 @@ BLEDevice BLEDeviceManager::available()
     {
         uint64_t timestamp_delta = timestamp - _peer_adv_mill[i];
         temp = &_peer_adv_buffer[i];
-        if ((timestamp_delta <= 2000) && (max_delta < timestamp_delta))
```

Comment:
@sandeepmistry I think some peripheral device will not give the scan response. Then those device will no be returned in available. It still has potential bugs. The below is I get from the BLE spec.

If the advertiser receives a SCAN_REQ PDU that contains its device address from a scanner allowed by the advertising filter policy it shall reply with a SCAN_RSP PDU on the same advertising channel index. After the SCAN_RSP PDU is sent or **if the advertising filter policy prohibited processing the SCAN_REQ PDU the advertiser** shall either move to the next used advertising channel index to send another ADV_SCAN_IND PDU, or close the advertising event.